### PR TITLE
🌱 Bump k8s libraries to v1.25

### PR DESCRIPTION
**Pull request by Fedosin**
**What this PR does / why we need it:**

This PR updates required libraries for cluster-api-operator to match kubernetes v1.25.

### CODE REVIEW

**Review by alexander-demicev — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.2.0
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.3.0
 
 require (
-	github.com/google/go-cmp v0.5.8
-	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.19.0
+	github.com/google/go-cmp v0.5.9
+	github.com/onsi/ginkgo/v2 v2.5.1
```
Comment (JoelSpeed) [inline go.mod:9]: Maybe bump to 2.6.1 while we are here?

File: go.mod [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+	k8s.io/api v0.25.3
+	k8s.io/apiextensions-apiserver v0.25.3
+	k8s.io/apimachinery v0.25.3
+	k8s.io/client-go v0.25.3
+	k8s.io/component-base v0.25.3
+	k8s.io/klog/v2 v2.80.1
+	k8s.io/utils v0.0.0-20221012122500-cfd413dd9e85
+	sigs.k8s.io/cluster-api v1.3.0
+	sigs.k8s.io/cluster-api/test v1.3.0
+	sigs.k8s.io/controller-runtime v0.13.1
```
Comment (JoelSpeed) [inline go.mod:22]: Is this on 1.25 or 1.26 deps?

File: hack/tools/go.mod [hunk 3]
```diff
@@ -1,11 +1,11 @@
 module sigs.k8s.io/cluster-api-operator/hack/tools
 
-go 1.17
+go 1.19
 
 require (
 	github.com/drone/envsubst/v2 v2.0.0-20210730161058-179042472c46
 	github.com/joelanford/go-apidiff v0.1.0
-	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/ginkgo/v2 v2.5.1
```
Comment (JoelSpeed) [inline hack/tools/go.mod:8]: 2.6.1?

File: test/e2e/e2e_suite_test.go [hunk 4]
```diff
@@ -114,7 +113,7 @@ func init() {
 
 func TestE2E(t *testing.T) {
 	RegisterFailHandler(Fail)
-	junitPath := filepath.Join(artifactFolder, fmt.Sprintf("junit.e2e_suite.%d.xml", config.GinkgoConfig.ParallelNode))
+	junitPath := filepath.Join(artifactFolder, fmt.Sprintf("junit.e2e_suite.%d.xml", GinkgoParallelProcess()))
 	junitReporter := reporters.NewJUnitReporter(junitPath)
 	RunSpecsWithDefaultAndCustomReporters(t, "capi-operator-e2e", []Reporter{junitReporter})
```
Comment (JoelSpeed) [inline test/e2e/e2e_suite_test.go:118]: I thought custom reporters were a thing of the past? I think ginkgo handles the parallel node thing for us now

File: go.mod (hunk 1 above)
Comment (Fedosin) [reply go.mod:9]: I bumped it to 2.7.0, as it it the latest released version.

File: go.mod (hunk 2 above)
Comment (Fedosin) [reply go.mod:22]: 1.25

**Comment by damdo** [comment]
@Fedosin do we know why the `apidiff` is failing? I'd be good to get this merged.

**Review by alexander-demicev — APPROVED** [review]
@damdo looks like this commit in apimachinery is causing failures https://github.com/kubernetes/apimachinery/commit/430b920312ca0fa10eca95967764ff08f34083a3, we can ignore it

/lgtm

File: test/e2e/e2e_suite_test.go (hunk 4 above)
Comment (Fedosin) [reply test/e2e/e2e_suite_test.go]: Yeah, you are right, I've fixed this part

**Comment by Fedosin** [comment]
Yeah, we can ignore apidiff output in this case

**Pull request merged**